src/emulator/functionsEmulator.ts

```diff
@@ -113,41 +92,34 @@ export class FunctionsEmulator implements EmulatorInstance {
     });
 
     hub.get("/", async (req, res) => {
-      res.send(
-        JSON.stringify(
-          await getTriggersFromDirectory(this.projectId, this.functionsDir, this.firebaseConfig),
-          null,
-          2
-        )
-      );
+      res.json({ status: "alive" });
     });
 
     // The URL for the function that the other emulators (Firestore, etc) use.
     // TODO(abehaskins): Make the other emulators use the route below and remove this.
-    const internalRoute = "/functions/projects/:project_id/triggers/:trigger_name";
+    const backgroundFunctionRoute = "/functions/projects/:project_id/triggers/:trigger_name";
 
     // The URL that the developer sees, this is the same URL that the legacy emulator used.
-    const externalRoute = `/:project_id/:region/:trigger_name`;
+    const httpsFunctionRoute = `/:project_id/:region/:trigger_name`;
 
     // A trigger named "foo" needs to respond at "foo" as well as "foo/*" but not "fooBar".
-    const functionRoutes = [
-      internalRoute,
-      `${internalRoute}/*`,
-      externalRoute,
-      `${externalRoute}/*`,
-    ];
+    const httpsFunctionRoutes = [httpsFunctionRoute, `${httpsFunctionRoute}/*`];
 
-    // Define a common handler function to use for GET and POST requests.
-    const handler: express.RequestHandler = async (req, res) => {
+    const backgroundHandler = async (req: express.Request, res: express.Response) => {
       const method = req.method;
-      const triggerName = req.params.trigger_name;
+      const triggerId = req.params.trigger_name;
 
-      logger.debug(`[functions] ${method} request to function ${triggerName} accepted.`);
+      logger.debug(`[functions] ${method} request to function ${triggerId} accepted.`);
 
       const reqBody = (req as RequestWithRawBody).rawBody;
-      const proto = reqBody ? JSON.parse(reqBody) : undefined;
+      const proto = JSON.parse(reqBody);
```

Comment:
JSON.parse(undefined) is not safe, so is this safe?

Reply:
It's purposely not safe, this endpoint will only be hit from our internal now (i.e. from Firestore Emulator), so if that emulator isn't passing a body that's an issue and we should explode.

Reply:
Ok maybe just add a `try/catch` to improve the error message because otherwise you'll get:

```
VM799:1 Uncaught SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at <anonymous>:1:6
```